plugin.audio.somafm/resources/lib/addon.py

```diff
@@ -0,0 +1,235 @@
+import os
+import shutil
+import sys
+import time
+import urllib.parse
+import urllib.request
+import xml.etree.ElementTree as ET
+
+import xbmc
+import xbmcaddon
+import xbmcgui
+import xbmcplugin
+import xbmcvfs
+from xbmcgui import ListItem
+from xbmcplugin import SORT_METHOD_GENRE, SORT_METHOD_LISTENERS, SORT_METHOD_UNSORTED
+
+from resources.lib.channel import Channel
+
+CHANNELS_FILE_NAME = "channels.xml"
+
+__addon__ = "SomaFM"
+__addonid__ = "plugin.audio.somafm"
+__version__ = "2.0.0"
+
+__ms_per_day__ = 24 * 60 * 60 * 1000
+
+
+def log(msg):
+    xbmc.log(str(msg), level=xbmc.LOGDEBUG)
+
+
+log(sys.argv)
+
+rootURL = "https://somafm.com/"
+tempdir = xbmcvfs.translatePath("special://home/userdata/addon_data/%s" % __addonid__)
+xbmcvfs.mkdirs(tempdir)
+
+LOCAL_CHANNELS_FILE_PATH = os.path.join(tempdir, CHANNELS_FILE_NAME)
+
+try:
+    plugin_url = sys.argv[0]
+    handle = int(sys.argv[1])
+    query = sys.argv[2]
+except:
```

Comment:
It's better to avoid using a bare `except` because it masks real errors in your code.